### PR TITLE
feat: add `blas/base/drotm-wasm`

### PR DESCRIPTION
**Pull request by aman-095**
Progresses #2039.

## Description

> What is the purpose of this pull request?

This pull request adds `wasm` support for `drotm`.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   progresses #2039 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by aman-095** [comment]
Hey, @kgryte, don't know why the test case inside `test.main.js` and `test.ndarray.js` for `N < 0` is giving an error.
```
	drotm.main( -1, x, 1, y, 1, param );
```

Error message:
```
# if provided an `N` parameter less than or equal to `0`, the `main` method leaves both input arrays unchanged
/home/aman/stdlib/lib/node_modules/@stdlib/strided/base/write-dataview/lib/ndarray.js:103
                view[ m ]( iv, data[ ix ], littleEndian );
                         ^

RangeError: Offset is outside the bounds of the DataView
    at DataView.setFloat64 (<anonymous>)
    at indexed (/home/aman/stdlib/lib/node_modules/@stdlib/strided/base/write-dataview/lib/ndarray.js:103:12)
    at writeDataView (/home/aman/stdlib/lib/node_modules/@stdlib/strided/base/write-dataview/lib/ndarray.js:296:9)
    at arrays2ptrs (/home/aman/stdlib/lib/node_modules/@stdlib/wasm/base/arrays2ptrs/lib/main.js:161:4)
    at Routine.drotm (/home/aman/stdlib/lib/node_modules/@stdlib/blas/base/drotm-wasm/lib/routine.js:176:9)
    at Routine.drotm (/home/aman/stdlib/lib/node_modules/@stdlib/blas/base/drotm-wasm/lib/routine.js:130:14)
    at Test.test (/home/aman/stdlib/lib/node_modules/@stdlib/blas/base/drotm-wasm/test/test.main.js:680:8)
    at Test.bound [as _cb] (/home/aman/stdlib/node_modules/tape/lib/test.js:58:32)
    at Test.run (/home/aman/stdlib/node_modules/tape/lib/test.js:77:10)
    at Test.bound [as run] (/home/aman/stdlib/node_modules/tape/lib/test.js:58:32)
```

File: lib/node_modules/@stdlib/blas/base/drotm-wasm/lib/module.js [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+*
+* // Define pointers (i.e., byte offsets) for storing three vectors:
+* var xptr = 0;
+* var yptr = N * bytesPerElement( dtype );
+* var pptr = 2 * N * bytesPerElement( dtype );
+*
+* // Write vector values to module memory:
+* drotm.write( xptr, oneTo( N, dtype ) );
+* drotm.write( yptr, ones( N, dtype ) );
+* drotm.write( pptr, zeros( 5, dtype ) );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/drotm-wasm/lib/module.js]: Using all zeros for the parameters is not a particularly "interesting" example.

**Comment by kgryte** [comment]
@aman-095 Looks like there is a bug in what we are passing to `strided2object`. Namely, it naively assumes that `N` is a nonnegative integer:

https://github.com/stdlib-js/stdlib/blob/7df58771f61965f3c819af3ea3b9376fedd86a86/lib/node_modules/%40stdlib/strided/base/strided2object/lib/main.js#L45

My guess is that we should update

https://github.com/stdlib-js/stdlib/blob/7df58771f61965f3c819af3ea3b9376fedd86a86/lib/node_modules/%40stdlib/strided/base/strided2object/lib/main.js#L76

to

```diff
- obj.length = N;
+ obj.length = ( N > 0 ) ? N : 0;
```

**Comment by kgryte** [comment]
This PR will need to incorporate various docs and build changes as made in other `blas/base/*-wasm` packages recently.

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @aman-095!

**Comment by kgryte** [comment]
The CI failure does not appear to be due to any issues with this PR. As such, will go ahead and merge.

**Pull request merged**